application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksGuildLeaveListener.java

```diff
@@ -0,0 +1,40 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.events.guild.member.GuildMemberJoinEvent;
+import net.dv8tion.jda.api.events.guild.member.GuildMemberRemoveEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import org.togetherjava.tjbot.commands.EventReceiver;
+
+/**
+ * Schedules a users bookmarks to be removed when leaving the guild. Also cancels the scheduled
+ * removal when rejoining before the bookmarks were removed.
+ */
+public final class BookmarksGuildLeaveListener extends ListenerAdapter implements EventReceiver {
```

Comment:
the name of the class isnt fully aligned with its features. right now it listenes to both:
- leave and
- join

but the name suggests only leave.

At this point, its probably better to rename the class to the feature, so `ExpireBookmarkListener` or similar

Reply:
Oh right i probably forgot that i had to cancel the scheduled removal for bookmarks and never noticed that again.

Reply:
Changed